pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -34,24 +35,27 @@ def __init__(self, amp_level: str = "O2") -> None:
         super().__init__()
         self.backend = AMPType.APEX
         self.amp_level = amp_level
+        self._connected = False
 
     def master_params(self, optimizer: Optimizer) -> _PARAMETERS:
         return amp.master_params(optimizer)
 
-    def connect(
-        self,
-        model: Module,
-        optimizers: List[Optimizer],
-        lr_schedulers: List[Any],
-    ) -> Tuple[Module, List[Optimizer], List[Any]]:
-        """Connects the precision plugin to the training process,
-        configures apex and reinits the schedulers
-        """
-        if model.device.type != "cuda":
-            return model, optimizers, lr_schedulers
-        model, optimizers = self.configure_apex(amp, model, list(optimizers), self.amp_level)
-        self.reinit_scheduler_properties(optimizers, lr_schedulers)
-        return model, optimizers, lr_schedulers
+    def pre_dispatch(self) -> None:
+        return super().pre_dispatch()
+
+    def dispatch(self, trainer: "pl.Trainer") -> None:
+        if not self._connected:
+            accelerator = trainer.accelerator
+            training_type_plugin = accelerator.training_type_plugin
+            lightning_module, optimizers = self.configure_apex(trainer.lightning_module, accelerator.optimizers)
+            self.reinit_scheduler_properties(optimizers, accelerator.lr_schedulers)
+            if not isinstance(training_type_plugin.model, LightningModule):
+                training_type_plugin.model.module.module = lightning_module
```

Comment:
this unwrapping + setter is doomed to fail haha
But I honestly don't know a better way right now that's quick to fix. 
So I suggest to put a TODO here, we definitely want to look into it, maybe the reorganization of moving the precision inside the trianing plugin, we could return the apex model back to the trainnig plugin.

Reply:
@awaelchli @tchaton if this is a temporary fix, could we mark these as private methods so we don't need to go through a full deprecation cycle for `dispatch` later on?

Reply:
Cleaned up !